### PR TITLE
Update substrate version used

### PR DESCRIPTION
**Pull request by ehildenb**
This updates the Substrate submodule to the most recent one which does _not_ remove the explicit `set_free_balance` function in the Wasm code. Any updates beyond that point will require us to change the way we invoke the functions to be consistent with the new generated code in the rule merger.

### CODE REVIEW

**Comment by ehildenb** [comment]
@hjorthjort I don't think those changes matter that much, probably those values are hashes of some chunks of the code. It does matter that the function names used in `search.py` actually match those in `src/polkadot-runtime.wat`.

File: search.py [hunk 1]
```diff
@@ -13,10 +13,8 @@
 sys.setrecursionlimit(1500000000)
 resource.setrlimit(resource.RLIMIT_STACK, (resource.RLIM_INFINITY, resource.RLIM_INFINITY))
 
-function_names = [ '$pallet_balances::Module<T_I>::set_free_balance::h143784e9433faed6'
-                 , '$pallet_balances::Module<T_I>::call_functions::h5c8befb10787dea0'
-                 , '$pallet_balances::Module<T_I>::storage_metadata::h082815e2817e5c19'
-                 , '$pallet_balances::Module<T_I>::module_constants_metadata::h487a5f31fed8642e'
+function_names = [ '$pallet_balances::Module<T_I>::set_free_balance::h434213974d3b5886'
+                 , '$pallet_balances::Module<T_I>::set_reserved_balance::h96655d728b3851f9'
```
Comment (hjorthjort) [inline search.py:17]: How about grabbing the function names dynamically? 
```
function_names = []
file = open("src/polkadot_runtime.wat")
for line in file:
    if "func $pallet_balances::Module<T_I>::set_free_balance" in line:
        function_names.append(line.split()[1])
    if "func $pallet_balances::Module<T_I>::set_reserved_balance" in line:
        function_names.append(line.split()[1])
close(file)
```

Comment (hjorthjort) [inline search.py:17]: How about grabbing the function names dynamically? 
```suggestion
function_names = []
file = open("src/polkadot_runtime.wat")
for line in file:
    if "func $pallet_balances::Module<T_I>::set_free_balance" in line:
        function_names.append(line.split()[1])
    if "func $pallet_balances::Module<T_I>::set_reserved_balance" in line:
        function_names.append(line.split()[1])
close(file)
```

Comment (ehildenb) [reply search.py]: Done.

**Pull request merged**